flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderElectionDriver.java

```diff
@@ -195,12 +196,14 @@ private class LeaderCallbackHandlerImpl extends KubernetesLeaderElector.LeaderCa
 
         @Override
         public void isLeader() {
-            leaderElectionEventHandler.onGrantLeadership(UUID.randomUUID());
+            leaderElectionEventHandler.onGrantLeadershipAsync(
+                    UUID.randomUUID(), Executors.newDirectExecutorService());
```

Comment:
Can `Executors.directExecutor()` be used here to replace `Executors.newDirectExecutorService()` 🤔

Reply:
Yes, good point. I'm gonna switch